src/Application/EntitySource.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseExport\Application;
+
+use Wikibase\DataModel\Entity\EntityDocument;
+
+interface EntitySource {
+
+	public function next(): ?EntityDocument;
+
+}
```

Comment:
This is the same interface as https://github.com/ProfessionalWiki/GND/blob/master/src/Domain/EntitySource.php

Unfortunately, we cannot reuse any of the implementations. The code in that repo is about import, not export :)